api/components/com_media/src/Controller/MediaController.php

```diff
@@ -0,0 +1,401 @@
+<?php
+/**
+ * @package         Joomla.API
+ * @subpackage      com_media
+ *
+ * @copyright   (C) 2021 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license         GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Component\Media\Api\Controller;
+
+\defined('_JEXEC') or die;
+
+use Joomla\CMS\Access\Exception\NotAllowed;
+use Joomla\CMS\Component\ComponentHelper;
+use Joomla\CMS\Filter\InputFilter;
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\MVC\Controller\ApiController;
+use Joomla\Component\Media\Administrator\Exception\FileExistsException;
+use Joomla\Component\Media\Administrator\Exception\InvalidPathException;
+use Joomla\Component\Media\Api\Helper\AdapterTrait;
+use Joomla\Component\Media\Api\Helper\MediaHelper;
+use Joomla\String\Inflector;
+use Tobscure\JsonApi\Exception\InvalidParameterException;
+
+/**
+ * Media web service controller.
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class MediaController extends ApiController
+{
+	use AdapterTrait;
+
+	/**
+	 * The content type of the item.
+	 *
+	 * @var    string
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $contentType = 'media';
+
+	/**
+	 * Query parameters => model state mappings
+	 *
+	 * @var  array
+	 */
+	private static $listQueryModelStateMap = [
+		'path'    => [
+			'name' => 'path',
+			'type' => 'STRING'
+		],
+		'url'     => [
+			'name' => 'url',
+			'type' => 'BOOLEAN'
+		],
+		'temp'    => [
+			'name' => 'temp',
+			'type' => 'BOOLEAN'
+		],
+		'content' => [
+			'name' => 'content',
+			'type' => 'BOOLEAN'
+		],
+	];
+
+	private static $itemQueryModelStateMap = [
+		'path'    => [
+			'name' => 'path',
+			'type' => 'STRING'
+		],
+		'url'     => [
+			'name' => 'url',
+			'type' => 'BOOLEAN'
+		],
+		'temp'    => [
+			'name' => 'temp',
+			'type' => 'BOOLEAN'
+		],
+		'content' => [
+			'name' => 'content',
+			'type' => 'BOOLEAN'
+		],
+	];
+
+	/**
+	 * The default view for the display method.
+	 *
+	 * @var    string
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $default_view = 'media';
+
+	/**
+	 * Display a list of files and/or folders.
+	 *
+	 * @return  static  A \JControllerLegacy object to support chaining.
+	 *
+	 * @throws  \Exception
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function displayList()
+	{
+		// Set list specific request parameters in model state.
+		$this->setModelState(self::$listQueryModelStateMap);
+
+		// Display files in specific path.
+		if ($this->input->exists('path'))
+		{
+			$this->modelState->set('path', $this->input->get('path', '', 'STRING'));
+		}
+
+		// Return files (not folders) as url's.
```

Comment:
plural, not ownership, correct english

```suggestion
		// Return files (not folders) as urls.
```